templates/docker/Dockerfile

```diff
@@ -36,15 +36,21 @@ FROM $BUILD_BASE AS builder
 ARG REPO=https://github.com/canonical/k8s-snap
 ARG BRANCH=main
 
+## Override the Kubernetes version from the branch.
+## Note(ben): We only have 1.30-release branches for k8s-snap right now.
+## For the rollout upgrades, we need to have a different minor version.
+## This is a temporary solution until we have a 1.31 release branch.
+ARG KUBERNETES_VERSION_OVERRIDE=""
```

Comment:
`ARG KUBERNETES_VERSION` should suffice, and no need for the extra comments. It's OK to have a build argument to specify the exact version if we want